spyder/app/tests/test_mainwindow.py

```diff
@@ -2209,6 +2209,47 @@ def test_varexp_magic_dbg(main_window, qtbot):
     assert shell._control.toHtml().count('img src') == 1
 
 
+@pytest.mark.slow
+@flaky(max_runs=3)
+@pytest.mark.skipif(os.name == 'nt', reason="Times out sometimes")
+def test_varexp_magic_dbg_locals(main_window, qtbot):
```

Comment:
I think you could add a test for this to the IPython console ones by running the `%varexp` magic manually while debugging and checking that a plot was generated on it.

I'd suggest to do that because the IPython console tests run much faster than the main window ones.